### ISSUE TITLE
S3 bucket creation fails

### ISSUE THREAD

**Issue by gadauto3** [terraformita/terraform-aws-serverless-app#3]
First off, thanks for supporting this example, it's exactly what I was looking for.

In the process of running the instructions, I get the following two errors:

```
│ Error: creating S3 bucket ACL for quick-bulldog-dev-gui: AccessControlListNotSupported: The bucket does not allow ACLs
│       status code: 400, request id: 9R...JJ, host id: pj...g=
│ 
│   with module.app.module.gui.aws_s3_bucket_acl.gui,
│   on ../../gui/main.tf line 14, in resource "aws_s3_bucket_acl" "gui":
│   14: resource "aws_s3_bucket_acl" "gui" {
│ 
|
│ Error: putting S3 policy: AccessDenied: Access Denied
│       status code: 403, request id: 9R...GW, host id: NB...0=
│ 
│   with module.app.module.gui.aws_s3_bucket_policy.gui,
│   on ../../gui/main.tf line 57, in resource "aws_s3_bucket_policy" "gui":
│   57: resource "aws_s3_bucket_policy" "gui" {
```

Googling the first error took me to these threads which I tried to no avail but maybe I'm missing something: 

1. https://stackoverflow.com/questions/76049290/error-accesscontrollistnotsupported-when-trying-to-create-a-bucket-acl-in-aws
2. https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/issues/223
3. https://stackoverflow.com/questions/76419099/access-denied-when-creating-s3-bucket-acl-s3-policy-using-terraform

Here's how I adjusted the aws_s3_bucket_acl resource based on 1.:
```
resource "aws_s3_bucket_acl" "gui" {
  bucket = aws_s3_bucket.gui.id
  acl    = "private"

  depends_on = [aws_s3_bucket_ownership_controls.s3_bucket_acl_ownership]
}

# Resource to avoid error "AccessControlListNotSupported: The bucket does not allow ACLs"
resource "aws_s3_bucket_ownership_controls" "s3_bucket_acl_ownership" {
  bucket = aws_s3_bucket.gui.id
  rule {
    object_ownership = "ObjectWriter"
  }
}
```
but still ended up with the following error:
```
│ Error: putting S3 policy: AccessDenied: Access Denied
│       status code: 403, request id: GRVFXE8FVHP90R0C, host id: 7DRfxGTBX05BdR9H6ezi/WtN4cCpGUbH5yde2M2dJKK2FPdzmwHbJcPiLSxNhf+P5/noj0ck/fM=
│ 
│   with module.app.module.gui.aws_s3_bucket_policy.gui,
│   on ../../gui/main.tf line 67, in resource "aws_s3_bucket_policy" "gui":
│   67: resource "aws_s3_bucket_policy" "gui" {
```
For the other two, I felt I made even less progress.

**Comment by kurianoff** [comment]
Hey @gadauto3 thank you for the request! I will look into it shortly, and will provide my update. I know the code in the module is a bit outdated, and AWS has changed the way it works wit bucket ACLs. So, please hold for a bit, I'll make the necessary change and will let you know.

**Comment by gadauto3** [comment]
Thanks @kurianoff , and this seems related: https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/issues/242

**Comment by kurianoff** [comment]
Hey @gadauto3 - the issue is resolved in version [0.1.8](https://github.com/terraformita/terraform-aws-serverless-app/releases/tag/v0.1.8)
Please check. 

**NOTE.** Newer version requires Terraform 1.3.0+ (previously it was working only under <=1.2.9).
